src/currency-field/CurrencyField.ts

```diff
@@ -232,220 +292,223 @@ export class CurrencyField extends OmniFormElement {
         }
     }
 
-    // Format the currency value when the component loses focus
-    async _blur(): Promise<void> {
-        const inputValue = this._inputElement?.value;
-
-        if (inputValue?.includes(this.fractionalSeparator)) {
-            // Split out the amount and fraction (cents) parts of the input value
-            const amountPart = this._parseAmount(inputValue.substring(0, inputValue.indexOf(this.fractionalSeparator)));
-
-            let fractionPart = this._parseFraction(inputValue.substring(inputValue.indexOf(this.fractionalSeparator) + 1));
-
-            const fractionalDifference = this.fractionalPrecision - fractionPart.length;
-            if (fractionalDifference > 0) {
-                for (let index = 0; index < fractionalDifference; index++) {
-                    fractionPart += '0';
-                }
-            }
+    // When the input is focussed set the default value to zero with the amount of cents dependent on the fractional precision.
+    _onFocusInput() {
+        const input = this._inputElement as HTMLInputElement;
+        if (!this.value) {
+            this.value = this._formatToCurrencyValue('0');
+        }
 
-            await this._formatToCurrency(amountPart as number).then((res) => {
-                this._inputElement!.value = res + (this.fractionalPrecision > 0 ? this.fractionalSeparator + fractionPart : '');
-            });
-        } else {
-            await this._formatToCurrency(this._parseAmount(inputValue as string) as number).then((res) => {
-                this._inputElement!.value = res;
-            });
+        if (input) {
+            setTimeout(() => {
+                input.selectionStart = input.selectionEnd = 10000;
```

Comment:
Should this not be based on the input value length instead of a harcoded value? What if the input has more than 10000 characters?
```suggestion
                input.selectionStart = input.selectionEnd = input.value?.length ?? 0;
```

Reply:
Implemented the suggested implementation